src/DynamoCoreWpf/Controls/ZoomBorder.cs

```diff
@@ -100,29 +100,34 @@ public Point GetTranslateTransformOrigin()
 
         public void SetTranslateTransformOrigin(Point2D p)
         {
-            var tt = GetTranslateTransform(child);
-            tt.X = p.X;
-            tt.Y = p.Y;
+            var tt = GetChildTranslateTransform();
+            if (tt.X != p.X || tt.Y != p.Y)
+            {
+                tt.X = p.X;
+                tt.Y = p.Y;
 
-            var st = GetScaleTransform(child);
-            NotifyViewSettingsChanged(tt.X, tt.Y, st.ScaleX);
+                var st = GetChildScaleTransform();
+                NotifyViewSettingsChanged(tt.X, tt.Y, st.ScaleX);
+            }
         }
 
         public void SetZoom(double zoom)
         {
-            var st = GetScaleTransform(child);
-            st.ScaleX = zoom;
-            st.ScaleY = zoom;
-
-            var tt = GetTranslateTransform(child);
-            NotifyViewSettingsChanged(tt.X, tt.Y, zoom);
+            var st = GetChildScaleTransform();
+            if (st.ScaleX != zoom || st.ScaleY != zoom)
+            {
+                st.ScaleX = zoom;
+                st.ScaleY = zoom;
+                var tt = GetTranslateTransform(child);
```

Comment:
GetChildTranslateTransform() ?
I saw you used this method at line 103

Reply:
Yes, could be used here. I may do a pass and use the child version of these methods.